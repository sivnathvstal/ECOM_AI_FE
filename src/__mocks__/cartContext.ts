import { Cart, CartProduct } from '../types';

export const mockCartItems: CartProduct[] = [
  {
    productId: 1,
    quantity: 2,
    product: {
      id: 1,
      title: 'Test Product 1',
      price: 10.99,
      description: 'Test description 1',
      category: 'electronics',
      image: 'test1.jpg',
      rating: { rate: 4.5, count: 100 },
    },
  },
  {
    productId: 2,
    quantity: 1,
    product: {
      id: 2,
      title: 'Test Product 2',
      price: 20.99,
      description: 'Test description 2',
      category: 'clothing',
      image: 'test2.jpg',
      rating: { rate: 4.0, count: 80 },
    },
  },
];

export const mockCart: Cart = {
  id: 1,
  userId: 1,
  date: '2023-01-01',
  products: [
    { productId: 1, quantity: 2 },
    { productId: 2, quantity: 1 },
  ],
};

export const mockCartContext = {
  cart: mockCart,
  cartItems: mockCartItems,
  isLoading: false,
  error: null,
  addItem: jest.fn(),
  updateItem: jest.fn(),
  removeItem: jest.fn(),
  totalItems: 3, // 2 + 1
  totalPrice: 42.97, // 10.99*2 + 20.99*1
};

export const mockEmptyCartContext = {
  cart: {
    id: 0,
    userId: 1,
    date: '2023-01-01',
    products: [],
  },
  cartItems: [],
  isLoading: false,
  error: null,
  addItem: jest.fn(),
  updateItem: jest.fn(),
  removeItem: jest.fn(),
  totalItems: 0,
  totalPrice: 0,
};

export const mockLoadingCartContext = {
  ...mockCartContext,
  isLoading: true,
};

export const mockErrorCartContext = {
  ...mockCartContext,
  error: 'Failed to fetch cart',
};

const defaultExport = {
  mockCartItems,
  mockCart,
  mockCartContext,
  mockEmptyCartContext,
  mockLoadingCartContext,
  mockErrorCartContext,
};
export default defaultExport; 