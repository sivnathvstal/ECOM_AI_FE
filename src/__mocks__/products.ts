import { Product, PaginationParams, ProductsResponse } from '../types';

export const mockProducts: Product[] = [
  {
    id: 1,
    title: 'Test Product 1',
    price: 10.99,
    description: 'Test description 1',
    category: 'electronics',
    image: 'test1.jpg',
    rating: { rate: 4.5, count: 100 },
  },
  {
    id: 2,
    title: 'Test Product 2',
    price: 20.99,
    description: 'Test description 2',
    category: 'clothing',
    image: 'test2.jpg',
    rating: { rate: 4.0, count: 80 },
  },
  {
    id: 3,
    title: 'Test Product 3',
    price: 15.99,
    description: 'Test description 3',
    category: 'electronics',
    image: 'test3.jpg',
    rating: { rate: 3.5, count: 60 },
  },
  {
    id: 4,
    title: 'Test Product 4',
    price: 25.99,
    description: 'Test description 4',
    category: 'jewelry',
    image: 'test4.jpg',
    rating: { rate: 4.8, count: 120 },
  },
];

export const mockPagination: PaginationParams = {
  page: 1,
  limit: 8,
  total: 4,
};

export const mockProductsResponse: ProductsResponse = {
  products: mockProducts,
  pagination: mockPagination,
};

export const mockCategories: string[] = [
  'electronics',
  'clothing',
  'jewelry',
]; 