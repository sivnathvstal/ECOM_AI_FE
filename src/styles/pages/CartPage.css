.cart-container {
  padding: 16px;
  margin-top: 16px;
}

.cart-title {
  margin-bottom: 16px;
}

.empty-cart {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 32px;
  text-align: center;
}

.empty-cart-icon {
  font-size: 4rem;
  margin-bottom: 16px;
  color: #757575; /* text.secondary */
}

.loader-container {
  display: flex;
  justify-content: center;
  padding: 32px;
}

.cart-summary {
  margin-top: 16px;
  padding: 16px;
}

.summary-item {
  display: flex;
  justify-content: space-between;
  padding: 8px 0;
}

.total {
  font-weight: bold;
  font-size: 1.2rem;
}

.checkout-button {
  margin-top: 16px;
}

.continue-shopping-button {
  margin-top: 16px;
} 