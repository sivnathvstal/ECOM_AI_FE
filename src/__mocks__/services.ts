import { mockProducts, mockProductsResponse, mockCategories } from './products';
import { mockCart, mockCartItems } from './cartContext';

// Product service mocks
export const mockProductService = {
  getProducts: jest.fn().mockResolvedValue(mockProductsResponse),
  getProductById: jest.fn().mockImplementation((id: number) => {
    const product = mockProducts.find(p => p.id === id);
    if (product) {
      return Promise.resolve(product);
    }
    return Promise.reject(new Error('Product not found'));
  }),
  getCategories: jest.fn().mockResolvedValue(mockCategories),
  searchProducts: jest.fn().mockResolvedValue(mockProductsResponse),
};

// Cart service mocks
export const mockCartService = {
  getCart: jest.fn().mockResolvedValue(mockCart),
  addToCart: jest.fn().mockResolvedValue(mockCart),
  updateCartItem: jest.fn().mockResolvedValue(mockCart),
  removeFromCart: jest.fn().mockResolvedValue(mockCart),
  getCartWithProducts: jest.fn().mockResolvedValue(mockCartItems),
}; 