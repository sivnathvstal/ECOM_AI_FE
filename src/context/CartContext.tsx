import React, { createContext, useState, useContext, useEffect, ReactNode } from 'react';
import { Cart, CartProduct } from '../types';
import { getCart, addToCart, updateCartItem, removeFromCart, getCartWithProducts } from '../services/cartService';

interface CartContextType {
  cart: Cart;
  cartItems: CartProduct[];
  isLoading: boolean;
  error: string | null;
  addItem: (productId: number, quantity: number) => Promise<void>;
  updateItem: (productId: number, quantity: number) => Promise<void>;
  removeItem: (productId: number) => Promise<void>;
  totalItems: number;
  totalPrice: number;
}

const CartContext = createContext<CartContextType | undefined>(undefined);

export const useCart = (): CartContextType => {
  const context = useContext(CartContext);
  if (!context) {
    throw new Error('useCart must be used within a CartProvider');
  }
  return context;
};

interface CartProviderProps {
  children: ReactNode;
}

export const CartProvider: React.FC<CartProviderProps> = ({ children }) => {
  const [cart, setCart] = useState<Cart>({
    id: 0,
    userId: 1,
    date: new Date().toISOString(),
    products: []
  });
  const [cartItems, setCartItems] = useState<CartProduct[]>([]);
  const [isLoading, setIsLoading] = useState<boolean>(true);
  const [error, setError] = useState<string | null>(null);

  const fetchCart = async () => {
    try {
      setIsLoading(true);
      const cartData = await getCart();
      setCart(cartData);
      const cartWithProducts = await getCartWithProducts();
      setCartItems(cartWithProducts);
      setError(null);
    } catch (err) {
      setError('Failed to fetch cart');
      console.error(err);
    } finally {
      setIsLoading(false);
    }
  };

  useEffect(() => {
    fetchCart();
  }, []);

  const addItem = async (productId: number, quantity: number) => {
    try {
      setIsLoading(true);
      const updatedCart = await addToCart(productId, quantity);
      setCart(updatedCart);
      await fetchCart(); // Refetch to get product details
    } catch (err) {
      setError('Failed to add item to cart');
      console.error(err);
    } finally {
      setIsLoading(false);
    }
  };

  const updateItem = async (productId: number, quantity: number) => {
    try {
      setIsLoading(true);
      const updatedCart = await updateCartItem(productId, quantity);
      setCart(updatedCart);
      await fetchCart(); // Refetch to get product details
    } catch (err) {
      setError('Failed to update cart item');
      console.error(err);
    } finally {
      setIsLoading(false);
    }
  };

  const removeItem = async (productId: number) => {
    try {
      setIsLoading(true);
      const updatedCart = await removeFromCart(productId);
      setCart(updatedCart);
      await fetchCart(); // Refetch to get product details
    } catch (err) {
      setError('Failed to remove item from cart');
      console.error(err);
    } finally {
      setIsLoading(false);
    }
  };

  // Calculate total items in cart
  const totalItems = cartItems.reduce((total, item) => total + item.quantity, 0);

  // Calculate total price
  const totalPrice = cartItems.reduce((total, item) => {
    return total + ((item.product?.price || 0) * item.quantity);
  }, 0);

  const value = {
    cart,
    cartItems,
    isLoading,
    error,
    addItem,
    updateItem,
    removeItem,
    totalItems,
    totalPrice
  };

  return <CartContext.Provider value={value}>{children}</CartContext.Provider>;
}; 