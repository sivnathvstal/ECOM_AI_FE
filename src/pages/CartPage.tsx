import React from 'react';
import { Link as RouterLink } from 'react-router-dom';
import {
  Container,
  Typography,
  Box,
  Button,
  Divider,
  CircularProgress,
  Paper
} from '@material-ui/core';
import { ShoppingBasket } from '@material-ui/icons';
import CartItemComponent from '../components/cart/CartItem';
import { useCart } from '../context/CartContext';
import './CartPage.css';

const CartPage: React.FC = () => {
  const { cartItems, isLoading, error, removeItem, totalItems, totalPrice } = useCart();

  if (isLoading) {
    return (
      <Container className="cart-container" data-testid="cart-loading">
        <Box className="loader-container">
          <CircularProgress />
        </Box>
      </Container>
    );
  }

  if (error) {
    return (
      <Container className="cart-container" data-testid="cart-error">
        <Typography color="error">
          {error}
        </Typography>
      </Container>
    );
  }

  if (cartItems.length === 0) {
    return (
      <Container className="cart-container" data-testid="empty-cart">
        <Paper>
          <Box className="empty-cart">
            <ShoppingBasket className="empty-cart-icon" />
            <Typography variant="h5" gutterBottom>
              Your cart is empty
            </Typography>
            <Typography variant="body1" color="textSecondary" paragraph>
              Looks like you haven't added any products to your cart yet.
            </Typography>
            <Button
              variant="contained"
              color="primary"
              component={RouterLink}
              to="/"
              data-testid="continue-shopping-button"
            >
              Continue Shopping
            </Button>
          </Box>
        </Paper>
      </Container>
    );
  }

  return (
    <Container className="cart-container" data-testid="cart-page">
      <Typography variant="h4" component="h1" className="cart-title" data-testid="cart-title">
        Shopping Cart
      </Typography>
      
      <Box data-testid="cart-items">
        {cartItems.map((item) => (
          <CartItemComponent
            key={item.productId}
            item={item}
            onRemove={removeItem}
          />
        ))}
      </Box>
      
      <Paper className="cart-summary" data-testid="cart-summary">
        <Typography variant="h6" gutterBottom>
          Order Summary
        </Typography>
        <Divider />
        <Box className="summary-item">
          <Typography variant="body1">
            Items ({totalItems})
          </Typography>
          <Typography variant="body1">
            ${totalPrice.toFixed(2)}
          </Typography>
        </Box>
        <Divider />
        <Box className="summary-item">
          <Typography variant="body1" className="total">
            Total
          </Typography>
          <Typography variant="body1" className="total" data-testid="cart-total">
            ${totalPrice.toFixed(2)}
          </Typography>
        </Box>
        <Button
          variant="contained"
          color="primary"
          fullWidth
          className="checkout-button"
          data-testid="checkout-button"
        >
          Checkout
        </Button>
        <Button
          variant="outlined"
          component={RouterLink}
          to="/"
          fullWidth
          className="continue-shopping-button"
          data-testid="continue-shopping"
        >
          Continue Shopping
        </Button>
      </Paper>
    </Container>
  );
};

export default CartPage; 