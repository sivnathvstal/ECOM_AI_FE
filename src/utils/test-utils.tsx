import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider, createTheme } from '@material-ui/core';
import { QueryClient, QueryClientProvider } from 'react-query';
import { CartProvider } from '../context/CartContext';

// Create a theme for testing
const theme = createTheme({
  palette: {
    primary: {
      main: '#3f51b5',
    },
    secondary: {
      main: '#f50057',
    },
  },
});

// Create a client for React Query
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

// All providers wrapper
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={theme}>
        <CartProvider>
          <BrowserRouter>{children}</BrowserRouter>
        </CartProvider>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// Re-export everything
export * from '@testing-library/react';

// Override render method
export { customRender as render }; 