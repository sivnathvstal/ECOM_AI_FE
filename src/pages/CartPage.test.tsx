import React from 'react';
import { render, screen, fireEvent, act } from '../utils/test-utils';
import CartPage from './CartPage';
import * as cartContext from '../context/CartContext';
import { mockCartContext, mockEmptyCartContext, mockLoadingCartContext, mockErrorCartContext } from '../__mocks__/cartContext';

// Mock the useCart hook
jest.mock('../context/CartContext', () => ({
  ...jest.requireActual('../context/CartContext'),
  useCart: jest.fn(),
}));

describe('CartPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with items in cart', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockCartContext);
    let container;
    await act(async () => {
      const rendered = render(<CartPage />);
      container = rendered.container;
    });
    expect(container).toMatchSnapshot();
  });

  it('renders loading state', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockLoadingCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    expect(screen.getByTestId('cart-loading')).toBeInTheDocument();
  });

  it('renders error state', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockErrorCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    expect(screen.getByTestId('cart-error')).toBeInTheDocument();
    expect(screen.getByText(mockErrorCartContext.error as string)).toBeInTheDocument();
  });

  it('renders empty cart state', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockEmptyCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    expect(screen.getByTestId('empty-cart')).toBeInTheDocument();
    expect(screen.getByText('Your cart is empty')).toBeInTheDocument();
    
    // Check if continue shopping button is present
    const continueShoppingButton = screen.getByTestId('continue-shopping-button');
    expect(continueShoppingButton).toBeInTheDocument();
    expect(continueShoppingButton).toHaveAttribute('href', '/');
  });

  it('renders cart items correctly', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    
    // Check if cart title is rendered
    expect(screen.getByTestId('cart-title')).toHaveTextContent('Shopping Cart');
    
    // Check if cart items are rendered
    mockCartContext.cartItems.forEach(item => {
      expect(screen.getByTestId(`cart-item-${item.productId}`)).toBeInTheDocument();
    });
    
    // Check if cart summary is rendered
    expect(screen.getByTestId('cart-summary')).toBeInTheDocument();
    expect(screen.getByTestId('cart-total')).toHaveTextContent(`$${mockCartContext.totalPrice.toFixed(2)}`);
  });

  it('calls removeItem when remove button is clicked', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    
    // Find and click the remove button for the first item
    const firstItem = mockCartContext.cartItems[0];
    const removeButton = screen.getByTestId(`cart-item-remove-${firstItem.productId}`);
    
    await act(async () => {
      fireEvent.click(removeButton);
    });
    
    // Check if removeItem was called with the correct productId
    expect(mockCartContext.removeItem).toHaveBeenCalledWith(firstItem.productId);
  });

  it('has working checkout button', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    
    // Find checkout button
    const checkoutButton = screen.getByTestId('checkout-button');
    expect(checkoutButton).toBeInTheDocument();
    expect(checkoutButton).toHaveTextContent('Checkout');
  });

  it('has working continue shopping button', async () => {
    (cartContext.useCart as jest.Mock).mockReturnValue(mockCartContext);
    await act(async () => {
      render(<CartPage />);
    });
    
    // Find continue shopping button
    const continueShoppingButton = screen.getByTestId('continue-shopping');
    expect(continueShoppingButton).toBeInTheDocument();
    expect(continueShoppingButton).toHaveTextContent('Continue Shopping');
    expect(continueShoppingButton).toHaveAttribute('href', '/');
  });
}); 