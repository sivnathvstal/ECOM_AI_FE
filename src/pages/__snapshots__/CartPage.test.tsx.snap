// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartPage renders correctly with items in cart 1`] = `
<div>
  <div
    class="MuiContainer-root cart-container MuiContainer-maxWidthLg"
    data-testid="cart-page"
  >
    <h1
      class="MuiTypography-root cart-title MuiTypography-h4"
      data-testid="cart-title"
    >
      Shopping Cart
    </h1>
    <div
      class="MuiBox-root MuiBox-root-1"
      data-testid="cart-items"
    >
      <div
        class="MuiPaper-root cart-item MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="cart-item-1"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <img
              alt="Test Product 1"
              class="cart-item-image"
              data-testid="cart-item-image-1"
              src="test1.jpg"
            />
          </div>
          <div
            class="MuiGrid-root cart-item-info MuiGrid-item MuiGrid-grid-xs-6"
          >
            <h6
              class="MuiTypography-root cart-item-title MuiTypography-subtitle1"
              data-testid="cart-item-title-1"
            >
              Test Product 1
            </h6>
            <div
              class="MuiBox-root MuiBox-root-2 cart-item-quantity"
              data-testid="cart-item-quantity-1"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Quantity: 
                2
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root cart-item-actions MuiGrid-item MuiGrid-grid-xs-3"
          >
            <p
              class="MuiTypography-root cart-item-price MuiTypography-body1"
              data-testid="cart-item-price-1"
            >
              $
              21.98
            </p>
          </div>
          <div
            class="MuiGrid-root cart-item-actions MuiGrid-item MuiGrid-grid-xs-1"
          >
            <button
              aria-label="delete"
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="cart-item-remove-1"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root cart-item MuiPaper-elevation1 MuiPaper-rounded"
        data-testid="cart-item-2"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <img
              alt="Test Product 2"
              class="cart-item-image"
              data-testid="cart-item-image-2"
              src="test2.jpg"
            />
          </div>
          <div
            class="MuiGrid-root cart-item-info MuiGrid-item MuiGrid-grid-xs-6"
          >
            <h6
              class="MuiTypography-root cart-item-title MuiTypography-subtitle1"
              data-testid="cart-item-title-2"
            >
              Test Product 2
            </h6>
            <div
              class="MuiBox-root MuiBox-root-3 cart-item-quantity"
              data-testid="cart-item-quantity-2"
            >
              <p
                class="MuiTypography-root MuiTypography-body2"
              >
                Quantity: 
                1
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root cart-item-actions MuiGrid-item MuiGrid-grid-xs-3"
          >
            <p
              class="MuiTypography-root cart-item-price MuiTypography-body1"
              data-testid="cart-item-price-2"
            >
              $
              20.99
            </p>
          </div>
          <div
            class="MuiGrid-root cart-item-actions MuiGrid-item MuiGrid-grid-xs-1"
          >
            <button
              aria-label="delete"
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="cart-item-remove-2"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiPaper-root cart-summary MuiPaper-elevation1 MuiPaper-rounded"
      data-testid="cart-summary"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
      >
        Order Summary
      </h6>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiBox-root MuiBox-root-4 summary-item"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Items (
          3
          )
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          $
          42.97
        </p>
      </div>
      <hr
        class="MuiDivider-root"
      />
      <div
        class="MuiBox-root MuiBox-root-5 summary-item"
      >
        <p
          class="MuiTypography-root total MuiTypography-body1"
        >
          Total
        </p>
        <p
          class="MuiTypography-root total MuiTypography-body1"
          data-testid="cart-total"
        >
          $
          42.97
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained checkout-button MuiButton-containedPrimary MuiButton-fullWidth"
        data-testid="checkout-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Checkout
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined continue-shopping-button MuiButton-fullWidth"
        data-testid="continue-shopping"
        href="/"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Continue Shopping
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
